webapp/src/components/App/Singlestudy/explore/Configuration/General/dialogs/ScenarioBuilderDialog/index.tsx

```diff
@@ -1,92 +1,46 @@
-import Rulesets from "./Rulesets";
-import Table from "./tabs/Table";
-import {
-  ACTIVE_SCENARIO_PATH,
-  getScenarioBuilderConfig,
-  ScenarioBuilderConfig,
-  TABS_DATA,
-} from "./utils";
-import ConfigContext from "./ConfigContext";
-import Thermal from "./tabs/Thermal";
+import Table from "./Table";
+import { getScenarioConfigByType, SCENARIOS, ScenarioType } from "./utils";
 import UsePromiseCond from "../../../../../../../common/utils/UsePromiseCond";
-import {
-  editStudy,
-  getStudyData,
-} from "../../../../../../../../services/api/study";
-import useEnqueueErrorSnackbar from "../../../../../../../../hooks/useEnqueueErrorSnackbar";
+import withAreas from "./withAreas";
+import usePromiseWithSnackbarError from "../../../../../../../../hooks/usePromiseWithSnackbarError";
 
 interface Props {
   study: StudyMetadata;
   open: boolean;
   onClose: VoidFunction;
-  nbYears: number;
 }
 
-function ScenarioBuilderDialog(props: Props) {
-  const { study, open, onClose, nbYears } = props;
-  const [currentTab, setCurrentTab] = useState(TABS_DATA[0][0]);
-  const [activeRuleset, setActiveRuleset] = useState("");
-  const [config, setConfig] = useState<ScenarioBuilderConfig>({});
-  const enqueueErrorSnackbar = useEnqueueErrorSnackbar();
-  const { t } = useTranslation();
-
-  const res = usePromise(async () => {
-    const config = await getScenarioBuilderConfig(study.id);
-    setConfig(config);
-
-    try {
-      const activeRuleset = await getStudyData<string>(
-        study.id,
-        ACTIVE_SCENARIO_PATH,
-      );
-
-      if (!config[activeRuleset]) {
-        throw new Error();
-      }
+// HOC that provides areas menu, for particular cases. (e.g thermals)
+const EnhancedTable = withAreas(Table);
 
-      setActiveRuleset(activeRuleset);
-    } catch {
-      setActiveRuleset("");
-    }
-  }, [study.id]);
-
-  const cxValue = useMemo(
-    () => ({
-      config,
-      setConfig,
-      reloadConfig: res.reload,
-      activeRuleset,
-      setActiveRuleset: (ruleset: string) => {
-        setActiveRuleset(ruleset);
+function ScenarioBuilderDialog({ study, open, onClose }: Props) {
+  const { t } = useTranslation();
+  const [selectedScenario, setSelectedScenario] = useState<ScenarioType>(
+    SCENARIOS[0],
+  );
 
-        editStudy(ruleset, study.id, ACTIVE_SCENARIO_PATH).catch((err) => {
-          setActiveRuleset("");
-          enqueueErrorSnackbar(
-            t(
-              "study.configuration.general.mcScenarioBuilder.error.ruleset.changeActive",
-            ),
-            err,
-          );
-        });
-      },
-      studyId: study.id,
-    }),
-    [activeRuleset, config, enqueueErrorSnackbar, res.reload, study.id, t],
+  const config = usePromiseWithSnackbarError(
+    () => getScenarioConfigByType(study.id, selectedScenario),
+    {
+      errorMessage: t(
+        "study.configuration.general.mcScenarioBuilder.noConfig.error",
+      ),
```

Comment:
Add dependencies: `deps: [study.id, selectedScenario]`

Reply:
its not needed, tab changes have explicit control over when the data fetching should occur.